runtime/string_functions.cpp

```diff
@@ -2537,7 +2537,8 @@ array<mixed> f$unpack(const string &pattern, const string &data) {
                 php_warning("Not enough data to unpack with format \"%s\"", pattern.c_str());
                 return result;
               }
-              value = f$strval(Long(*(long long *)(data.c_str() + data_pos)));
+              long long value_ll = *(long long *)(data.c_str() + data_pos);
```

Comment:
```suggestion
              long long value_ll = *reinterpret_cast<const long long *>(data.c_str() + data_pos);
```

Reply:
What's wrong with the original code?)

Reply:
1. C-style cast here silently discards const, which is absolutely not needed and potentially dangerous. One of the reason to avoid C-style casts is they perform sequence of different casts until the expression is casted even if you need particular one. It happens implicitly and you can't see, for example, that it has dropped const.
2. In general, there's no reason to use C-style casts in C++ code. C++ style casts are more explicit, easier for search and checked by the compiler.